core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala

```diff
@@ -133,6 +133,28 @@ private[spark] class ShuffleMapStage(
   }
 
   /**
+   * Removes all shuffle outputs associated with this host. Note that this will also remove
+   * outputs which are served by an external shuffle server (if one exists), as they are still
+   * registered with this execId.
+   */
+  def removeOutputsOnHost(host: String): Unit = {
```

Comment:
This function is pretty overlap in code with `removeOutputsOnExecutor`, How about combine them as:
```
  def removeOutputsByFilter(f: (BlockManagerId) => Boolean): Unit = {
    ......
      val newList = prevList.filterNot(m => f(m))
    ......
  }
```
And in DAGScheduler, we can simply pass in the filter functions.